moveit_ros/moveit_servo/tests/servo_ros_fixture.hpp

```diff
@@ -0,0 +1,134 @@
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of PickNik Inc. nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author    : V Mohammed Ibrahim
+   Desc      : Resources used by servo ROS integration tests
+   Title     : servo_ros_fixture.hpp
+   Project   : moveit_servo
+   Created   : 07/23/2023
+*/
+
+#include <gtest/gtest.h>
+#include <moveit_msgs/msg/servo_status.hpp>
+#include <moveit_msgs/srv/servo_command_type.hpp>
+#include <moveit_servo/utils/datatypes.hpp>
+#include <rclcpp/client.hpp>
+#include <rclcpp/node.hpp>
+#include <rclcpp/publisher.hpp>
+#include <rclcpp/qos.hpp>
+#include <sensor_msgs/msg/joint_state.hpp>
+
+class ServoRosFixture : public testing::Test
+{
+protected:
+  void statusCallback(const moveit_msgs::msg::ServoStatus::ConstSharedPtr msg)
+  {
+    status_ = static_cast<moveit_servo::StatusCode>(msg->code);
+    status_count_++;
+  }
+
+  void jointStateCallback(const sensor_msgs::msg::JointState::ConstSharedPtr msg)
+  {
+    std::lock_guard<std::mutex> state_guard(joint_state_mutex);
+    joint_states_ = *msg;
+    state_count_++;
+  }
+
+  void SetUp()
+  {
+    executor_->add_node(servo_test_node_);
+    executor_thread_ = std::thread([this]() { executor_->spin(); });
+  }
+
+  void TearDown()
+  {
+    executor_->cancel();
+    if (executor_thread_.joinable())
+    {
+      executor_thread_.join();
+    }
+  }
+
+  void waitForService()
+  {
+    auto logger = servo_test_node_->get_logger();
+    while (!switch_input_client_->service_is_ready())
+    {
+      if (!rclcpp::ok())
+      {
+        RCLCPP_ERROR(logger, "Interrupted while waiting for the service. Exiting.");
+      }
```

Comment:
This appears to not actually exit

Reply:
I think you could just add `std::exit(EXIT_FAILURE);`. That would cause whatever test to fail

Reply:
Thanks for the catch!
There was originally a `std::exit()` there, but got deleted sometime during the changes.